cloudify_rest_client/blueprints.py

```diff
@@ -225,6 +262,48 @@ def upload(self,
         finally:
             shutil.rmtree(tempdir)
 
+    def validate(self,
+                 path,
+                 entity_id,
+                 visibility=VisibilityState.TENANT,
+                 progress_callback=None,
+                 skip_size_limit=True):
+        """
+        Validates a blueprint with Cloudify's manager.
+
+        :param path: Main blueprint yaml file path.
+        :param entity_id: Id of the uploaded blueprint.
+        :param visibility: The visibility of the blueprint, can be 'private',
+                           'tenant' or 'global'.
+        :param progress_callback: Progress bar callback method
+        :param skip_size_limit: Indicator whether to check size limit on
+                           blueprint folder
+        :return: Created response.
+
+        Blueprint path should point to the main yaml file of the response
+        to be uploaded. Its containing folder will be packed to an archive
+        and get uploaded to the manager.
+        Validation is basically an upload without storage part being done.
+        """
+        tempdir = tempfile.mkdtemp()
+        try:
+            tar_path = utils.tar_blueprint(path, tempdir)
+            if not skip_size_limit and os.path.getsize(tar_path) > 30000000:
+                raise Exception('Blueprint folder exceeds 30 MB, '
+                                'move some resources from the blueprint '
+                                'folder to an external location or upload'
+                                ' the blueprint folder as a zip file.')
+            application_file = os.path.basename(path)
```

Comment:
The above logic is similar to what we have in upload method, cannot we this logic to share method called `_validte_blueprint_size` for example ?

Reply:
You are right; fixed that.